spec/tictactoe_spec.rb

```diff
@@ -1,31 +1,131 @@
 require 'tictactoe'
+require 'ai_basic'
+require 'game_rules'
 
 describe TicTacToe do
-  let (:game_io) { double }
-  let (:player1) { double }
-  let (:player2) { double }
-  let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
-
-  it "runs the main loop " do
-    expect(test_ttt.game_loop).to receive(:game_loop).once
-    expect(game_io).to receive(:puts_message)
-    expect(game_io).to receive(:get_input).and_return("N")
-    test_ttt.play_game
+  let(:rules) { double }
+  let(:board) { double }
+  let(:players) { double }
+  let(:state) { double }
+  let(:ai_basic) { AiBasic.new(GameRules.new) }
+  Player = Struct.new(:mark, :type)
```

Comment:
You already define this in a lib file somewhere .. does it make sense to expose it so you can use it here? Or is that strictly internal?

If it does _not_ make sense to expose it and you just need something that response to these two methods, an rspec `double(method_1: :value, method_2: :value_2)` is sufficient.

There is also `OpenStruct` which you can use super-duper dynamically (scary, but convenient, which usually means it'll bite you later). But it's there if you're feeling adventurous - google it.